### DISCUSSION TITLE
Exploring feasibility of using Tiny Core Linux instead of Buildroot

### DISCUSSION THREAD

**Discussion by Firminator** [PartialVolume/shredos.x86_64#229] [Ideas]
TCL was just updated[0] and it seems a perfect fit for ShredOS having under 10s boottime, running from RAM, recent 6.6.8 kernel, 32bit and 64bit architecture support[1][2], includes ncurses[3], readline[3], and a bunch of so-called extensions can be added and/or made persistent including sg3-tools, 'secure-erase', smartmontools, udisk, (and firmware blobs for video, NIC, RAID controller etc. which should work in our favor to support a wide range of devices).[4]
They have 3 different versions[5]:
* Core (17MB download)
* TinyCore (23 MB)
* CorePlus (248MB)

Unfortunately nwipe is not available as an extension (yet). They have a wiki entry how to create an extension although it's quite an undertaking[6].

Just throwing this out to see what opinions others have about TCL.

[0] https://www.theregister.com/2024/03/04/tiny_core_linux_15/
[1] http://tinycorelinux.net/15.x/x86/release/
[2] http://tinycorelinux.net/15.x/x86_64/release/
[3] http://tinycorelinux.net/15.x/x86_64/release/src/toolchain/
[4] http://tinycorelinux.net/15.x/x86_64/tcz/
[5] http://tinycorelinux.net/downloads.html
[6] https://wiki.tinycorelinux.net/doku.php?id=wiki:creating_extensions

**Comment by PartialVolume** [comment]
I think the current build of ShredOS is equivalent to TCL CorePlus mainly because the DRM and AMD, Intel & Nvidia drivers take up over 100MB of the 248MB of disc space. Of course ShredOS doesn't have a desktop although I think with minimal code it could be added because those graphics drivers are already present.

Looks interesting although I'm too invested in adding new features to nwipe and keeping ShredOS updated to spend any serious time on it at the moment.

The concept of loading apps via the app browser is interesting, made me think of a new option for ShredOS. At boot time it examines the USB for /usr/bin/nwipe and if present uses that instead of the built in version. This would allow a sort of repository of upto date nwipes to be applied to a existing ShredOS. I'm not sure if that's a good idea or not?
